### PR TITLE
add Messenger.Init()

### PR DESCRIPTION
**Pull request by adambabik**
This change adds `Messenger.Init()` method which based on the stored chats and contacts initializes Whisper filters.

It also contains some changes cleaning up term `filter` and `chat`.

Todo:
- [ ] add unit tests for `Messenger.Init()`

### CODE REVIEW

File: messenger.go [hunk 1]
```diff
@@ -317,6 +318,55 @@ func NewMessenger(
...
+	}
+	for _, chat := range chats {
+		if !chat.Active {
+			continue
+		}
+		if chat.ChatType == ChatTypePublic {
+			publicChatIDs = append(publicChatIDs, chat.ID)
+		} else if chat.ChatType == ChatTypeOneToOne {
+			publicKeys = append(publicKeys, chat.PublicKey)
+		}
```
Comment (cammellos) [inline messenger.go]: Forgot to mention sorry, we also need to create a filter for each group chat member, `ChatMember.ID` is the hex encoded public key

File: contact.go [hunk 2]
```diff
@@ -31,3 +44,33 @@ type Contact struct {
...
+
+func (c Contact) HasBeenAdded() bool {
+	return existsInStringSlice(c.SystemTags, contactRequestReceived)
+}
+
+func (c Contact) IsBlocked() bool {
+	return existsInStringSlice(c.SystemTags, contactBlocked)
+}
+
+// exists checks if a string is in a set.
```
Comment (PombeirP) [inline contact.go]:
```suggestion
// existsInStringSlice checks if a string is in a set.
```

File: messenger.go [hunk 3]
```diff
@@ -317,6 +318,55 @@ func NewMessenger(
...
+	// TODO: Get only active chats by the query.
+	chats, err := m.Chats()
+	if err != nil {
+		return err
+	}
+	for _, chat := range chats {
+		if !chat.Active {
+			continue
+		}
+		if chat.ChatType == ChatTypePublic {
```
Comment (PombeirP) [inline messenger.go]: Why not use a `switch` statement instead?

File: messenger.go (hunk 1 above)
Comment (adambabik) [reply messenger.go]: Added.

**Comment by adambabik** [comment]
Sorry, I forgot about unit tests. Will add asap.

File: messenger.go (hunk 1 above)
Comment (cammellos) [reply messenger.go]: looks good, thanks, just a note that you might have duplicates, but if we are fine with that and they are handled properly then no probs

Comment (adambabik) [reply messenger.go]: It is. If a filter of the same type for a given public key exists, it will be returned early without creating a duplicate.

**Pull request merged**